net/dns/resolver/forwarder.go

```diff
@@ -419,6 +448,20 @@ func (f *forwarder) sendDoH(ctx context.Context, urlBase string, c *http.Client,
 	req.Header.Set("Accept", dohType)
 	req.Header.Set("User-Agent", "tailscaled/"+version.Long())
 
+	// If this is a Control D request, attach the hostname, mac address and IP address
+	// this allows for additional DNS analytics and client specific DNS policies
+	if strings.HasPrefix(urlBase, publicdns.ControlDBase) {
+		if f.dohClientHostname != "" {
+			req.Header.Set("X-Cd-Host", f.dohClientHostname)
+		}
+		if f.dohClientMac != "" {
+			req.Header.Set("X-Cd-Mac", f.dohClientMac)
+		}
+		if f.dohClientIP != "" {
+			req.Header.Set("X-Cd-IP", f.dohClientIP)
```

Comment:
lowercase "p" in "Ip" (otherwise Set will just do it for you, allocating)

Reply:
fixed